readthedocs/core/resolver.py

```diff
@@ -204,6 +188,69 @@ def resolve(
         )
         return urlunparse((protocol, domain, path, '', query_params, ''))
 
+    def _get_canonical_project_data(self, project):
+        """
+        Returns a tuple with (project, subproject_slug) from the canonical project of `project`.
+
+        We currently support more than 2 levels of nesting subprojects and translations,
+        but we only serve 2 levels to avoid sticking in the loop.
+        This means, we can have the following cases:
+
+        - The project isn't a translation or subproject
+
+          We serve the documentation from the domain of the project itself
+          (main.docs.com/).
+
+        - The project is a translation of a project
+
+          We serve the documentation from the domain of the main translation
+          (main.docs.com/es/).
+
+        - The project is a subproject of a project
+
+          We serve the documentation from the domain of the super project
+          (main.docs.com/projects/subproject/).
+
+        - The project is a translation, and the main translation is a subproject of a project, like:
+
+          - docs.docs.com
+          - api.docs.com (subproject of ``docs``)
+          - api-es.docs.com (translation of ``api``, and current project to be served)
+
+          We serve the documentation from the domain of the super project
+          (docs.docs.com/projects/api/es/).
+
+        - The project is a subproject, and the superproject is a translation of a project, like:
+
+          - docs.docs.com/
+          - docs-es.docs.com/ (translation of ``docs``)
+          - api-es.docs.com/ (subproject of ``docs-es``, and current project to be served)
+
+          We serve the documentation from the domain of the super project (the translation),
+          this is docs-es.docs.com/projects/api-es/es/.
+
+          Note: we aren't going to support this case for now.
+
+        In summary: If the project is a subproject,
+        we don't care if the superproject is a translation,
+        we always serve from the domain of the superproject.
+        If the project is a translation,
+        we need to check if the main translation is a subproject.
+        """
+        main_project = project
+        subproject_slug = None
+
+        main_language_project = main_project.main_language_project
+        if main_language_project:
+            main_project = main_language_project
+
+        relation = main_project.get_parent_relationship()
+        if relation:
+            main_project = relation.parent
+            subproject_slug = relation.alias
+
+        return (main_project, subproject_slug)
+
     def _get_canonical_project(self, project, projects=None):
```

Comment:
we can re-use the above method here if we don't't want to support more than 2 levels of relationships (thing that we already don't support actually...)

Reply:
I don't understand this comment? re-use what method?

Reply:
This method is basically the same as `_get_canonical_project_data` (when checking till 2 levels of relationships)